### ISSUE TITLE
switch UUID PIDS but preserve old PIDS

### ISSUE THREAD

**Issue by davidschober** [avalonmediasystem/avalon#836]
The new Hydra/Fedora style PIDS will supplant the old "avalon:333" style PIDs (we'll want to retain the latter). We need to figure out where to stuff the Fedora 3 style PIDs so that people can still access them. 

Fedora3 style PIDs are stored in the Dublin Core datastream. 

This might be for free.
URLs using the Fedora 3 style PID will not work during this scheme. This needs to be evaluated by POs.
### Done looks like
- [x] Use the new stuff.

Documentation Story: #1503

**Comment by joncameron** [comment]
David Schober added a comment - 21/Apr/16 10:45 AM
Notes from 04/21

* The ID should be there when the record is created
* Default is uuid on the ground completely randomly generated
* One level of override is add prefix it and another is to monkey patch.
* we already provide a method for overriding
* possibly have a prefix to the UID in avalon config

**Comment by davidschober** [comment]
We will have to customize our gem.

**Comment by davidschober** [comment]
Part of this is related to migration gem. That is where we will actually create them.

**Comment by jlhardes** [comment]
Need to review RDF property used to define F3 pid in context of other identifiers to make sure we're expressing things in a way we can understand.

**Comment by davidschober** [comment]
Note, this is dependent upon #1310

**Comment by davidschober** [comment]
@joncameron let's rework this into a documentation gem.

**Comment by joncameron** [comment]
I'll talk with @jlhardes to clarify this issue.

**Comment by jlhardes** [comment]
Slack conversation between @mbklein and @jlhardes about RDF property to use for F3 identifiers in F4:

Julie Hardesty [1:38 PM]  
Hey there! Do you have a sec for a couple questions about F3 to F4 properties with Avalon objects? It has to do with https://github.com/avalonmediasystem/avalon/issues/939 and verifying that the old F3 pid is being preserved.

[1:41]  
there's also another issue that deal with documenting this info for developers - i'm looking for it now and we were just discussing it this morning

[1:43]  
oh, this one: https://github.com/avalonmediasystem/avalon/issues/836

[1:44]  
i have a meeting from 2-3pm so if you don't have time now i'll check back in later on

[1:46]  
i'm thinking that all "identifier" properties in the different classes connected to this predicate review issue: https://github.com/avalonmediasystem/avalon/issues/1029 reflect the F3 pid being brought over. Is that right? (edited)

Michael Klein [2:24 PM]  
Oh hey. I just finished lunch and saw this. I'll look it over and we can be in touch after your meeting.

Julie Hardesty [3:15 PM]  
No problem, I'm finally done with my meeting. Are you around?

Julie Hardesty [3:46 PM]  
ok, i have another meeting in about 15 minutes. i'll try again when that one is done and see if you are available. TAG! :running:

Michael Klein [3:50 PM]  
Oops. Sorry. I’m here!

[3:51]  
Here’s my issue. We can’t just make all of these `dc:identifier`s because we need some way to provide context. Like, “Oh, hey, these are all the different identifiers this thing is known by but _this one_ is the old Fedora 3 PID."

[3:51]  
Because otherwise we lose data resolution.

Julie Hardesty [3:57 PM]  
yes, i agree. i have to get into my other meeting but i can get back on when that's over.

Julie Hardesty [4:27 PM]  
OK, i am back!

[4:30]  
@mbklein ping

Julie Hardesty [4:35 PM]  
i see we have an identifier pegged for MediaObject (loc:local) and MasterFile (also loc:local) but we don't have anything on Derivative, so that makes it less clear to me that we're using those to bring over F3 avalon pids.

[4:36]  
so do we need a different identifier property to use for avalon pids from F3 that will only be there if an object is migrated from F3 to F4?

Michael Klein [4:43 PM]  
Yes. I’ve been using a made-up one _only_ during migration, but it’d be good to have a real one.

[4:43]  
Or just have an identifier prefixing paradigm, but that would mess up linked data interoperability.

Julie Hardesty [4:44 PM]  
so are the loc:local properties on masterfile and mediaobject for the F4 identifiers?

Michael Klein [4:44 PM]  
Umm… I can’t remember. Hang on.

[4:47]  
It looks like the F3 identifier might be getting stashed in there. But I don’t know why it’s not happening for derivatives.

Julie Hardesty [4:47 PM]  
ok, so could we use loc:local on Derivative for the F3 identifiers as well then?

Michael Klein [4:48 PM]  
Yes, but only if we don’t use loc:local for anything else. Like actual local identifiers.

Julie Hardesty [4:48 PM]  
right, and that's actually a problem when mapping MODS.

[4:49]  
but if we can use the same property for F3 identifiers on mediaobject, masterfile, and derivative, i can look for another property to use that isn't loc:local

Michael Klein [4:51 PM]  
Yes. We can absolutely use the same property everywhere.

[4:51]  
The one I made up to use during migration (to remap relationships) is `http://avalonmediasystem.org/ns/migration#migratedFrom`

Julie Hardesty [4:52 PM]  
ok, i am going to find a different property than loc:local to use for F3 identifiers

Michael Klein [4:52 PM]  
But that’s not standard.

[4:52]  
Obviously. (edited)

Julie Hardesty [4:52 PM]  
there should be something we can use - it might not actually mean a migrated identifier but some kind of identifier property that already exists and we can pick out as the one that means fedora3 identifier.

[4:52]  
heck, can we call up a fedora property?

Michael Klein [5:02 PM]  
I don’t know.

Julie Hardesty [5:03 PM]  
well, the migration-utils project shows that fedora 3's namespace is not a published namespace, so they made something up for the PID as well: https://github.com/fcrepo4-exts/migration-utils

[5:03]  
i'll keep looking

[5:07]  
there are identifier properties from a vocab for conversions: http://lov.okfn.org/dataset/lov/terms?q=identifier&type=property&tag=RDF
 
[5:11]  
something like that might work

Michael Klein [5:13 PM]  
Ooh `conversion:source_identifier` looks promising

Julie Hardesty [5:14 PM]  
source identifier is supposed to be for the organization providing the dataset

[5:14]  
could dataset_identifier work?

Michael Klein [5:14 PM]  
Oh.

[5:14]  
Maybe?

Julie Hardesty [5:14 PM]  
it's more like the thing we're trying to identify

Michael Klein [5:14 PM]  
Yeah

Julie Hardesty [5:14 PM]  
the original thing

[5:14]  
it's not a dataset but it's the original thing

Michael Klein [5:14 PM]  
I still feel like we’re shoehorning it.

Julie Hardesty [5:14 PM]  
and then the predicate could be "conversion"

[5:14]  
maybe

Michael Klein [5:15 PM]  
I need to book out of here, but I’m prepared to keep talking about this tomorrow. :slightly_smiling_face:

Julie Hardesty [5:15 PM]  
yeah, i need to get out too but i think there's something we can use

[5:15]  
thanks!

Michael Klein [5:15 PM]  
:+1:


----- Today January 6th, 2017 -----
Julie Hardesty [8:19 AM]  
@mbklein what about owl:sameAs? I don't think Avalon is using owl properties for anything else. Even though the avalon pid from F3 doesn't technically exist anymore, it can still be considered a URI reference referring to the same thing. I'm still looking but I kind of like that one.

Julie Hardesty [8:51 AM]  
Here's another thing I've found: the Ontology for Media Resources 1.0 has an identifier property and specifies that if it's a legacy identifier then a URI should be minted (https://www.w3.org/TR/mediaont-10/#property-definition). Using that would make the property ma-ont:identifier with a minted URI containing the Avalon pid that goes to nowhere since it doesn't exist anymore. We could do a similar thing with owl:sameAs.
